### ISSUE TITLE
Ack! No DDC/CI support on Odyssey G9!

### ISSUE THREAD

**Issue by kcorey** [haimgel/display-switch#43]
Oh, I am sad.

It's unfair reporting this as an 'issue' with display-switch...

I got an early Christmas present: a LC29G95TSSUXEN monitor.  It's *great*...with one exception: it doesn't seem to support DDC/CI.

From my MacBook, using ddcctl, I get this sort of thing:
$ ./ddcctl -d 1 -i 18
D: NSScreen #190590103 (5120x1440 0°) 109.00 DPI
I: found 1 external display
I: polling display 1's EDID
I: got edid.name: LC49G95T
I: got edid.serial: H4ZN901744
D: action: i: 18
D: setting VCP control #96 => 18

And then nothing happens (This happens with the display number (the -ii parm) from 1..18.)

In display_switch, I get:
15:30:53 [ERROR] Failed to get current input for display '0'
15:31:03 [DEBUG] (1) display_switch::app: Detected device change. Removed device: "0bda:5411"
15:31:03 [DEBUG] (1) display_switch::app: Detected device change. Removed device: "05ac:1006"
15:31:03 [DEBUG] (1) display_switch::app: Detected device change. Removed device: "05ac:0221"
15:31:03 [DEBUG] (1) display_switch::app: Detected device change. Removed device: "046d:c52f"
15:31:03 [DEBUG] (1) display_switch::app: Detected device change. Removed device: "0bda:0411"
15:31:05 [DEBUG] (1) display_switch::app: Detected device change. Added device: "0bda:0411"
15:31:05 [INFO] Detected device we're looking for "0bda:0411"
15:31:05 [INFO] Display '0' set to 0xf
15:31:05 [DEBUG] (1) display_switch::app: Detected device change. Added device: "0bda:5411"
15:31:06 [DEBUG] (1) display_switch::app: Detected device change. Added device: "05ac:1006"
15:31:06 [DEBUG] (1) display_switch::app: Detected device change. Added device: "046d:c52f"
15:31:07 [DEBUG] (1) display_switch::app: Detected device change. Added device: "05ac:0221"
15:31:12 [DEBUG] (1) display_switch::app: Detected device change. Removed device: "0bda:5411"
15:31:12 [DEBUG] (1) display_switch::app: Detected device change. Removed device: "05ac:1006"
15:31:12 [DEBUG] (1) display_switch::app: Detected device change. Removed device: "046d:c52f"
15:31:12 [DEBUG] (1) display_switch::app: Detected device change. Removed device: "05ac:0221"
15:31:12 [DEBUG] (1) display_switch::app: Detected device change. Removed device: "0bda:0411"
15:31:14 [DEBUG] (1) display_switch::app: Detected device change. Added device: "0bda:0411"
15:31:14 [INFO] Detected device we're looking for "0bda:0411"
15:31:14 [INFO] Display '0' set to 0xf
15:31:14 [DEBUG] (1) display_switch::app: Detected device change. Added device: "0bda:5411"
15:31:14 [DEBUG] (1) display_switch::app: Detected device change. Added device: "05ac:1006"
15:31:15 [DEBUG] (1) display_switch::app: Detected device change. Added device: "046d:c52f"
15:31:15 [DEBUG] (1) display_switch::app: Detected device change. Added device: "05ac:0221"

So you can see me switching 4 times here.

That's me fairly well out of the switching game, I'm afraid, unless someone can come up with a way to shake this monitor into submission, or have an alternative control scheme.

So sad.

**Comment by haimgel** [comment]
Oh, this is indeed sad! This looks like a very nice monitor. The only things I can suggest are this:
1. Try using a different port _type_. E.g. if you're using DisplayPort/USB-C, try via HDMI, and vice-versa.
2. Try a different computer/os, if possible, to rule out a software issue.

But, my (brief) Google search did turn up similar complaints, so you might be out of luck here :-(

**Comment by kcorey** [comment]
Wow...that works!

You're a miracle worker.

Firstly I had to update my fork to reflect all the recent changes in your fork (https://ardalis.com/syncing-a-fork-of-a-github-repository-with-upstream/).

Then a quick recompile.

I updated my .ini file to reflect the new syntax.
```
    usb_device = "0bda:0411"
    on_usb_connect = "Hdmi1"
    on_usb_disconnect = "DisplayPort1"
```

I connected my Mac to the G9 with this cable:
https://www.amazon.co.uk/gp/product/B07WXGKKBW/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1
(This is only capable of 3840x1080 @60Hz, but I'll take it!)

I connect my work Mac to the G9 with a DisplayPort cable, and no software on it.

Viola!  Switching goodness.

You rock!
